lib/Sema/CSDiag.cpp

```diff
@@ -2387,13 +2387,13 @@ bool FailureDiagnosis::diagnoseGeneralMemberFailure(Constraint *constraint) {
 
   // If this is a tuple, then the index needs to be valid.
   if (auto tuple = baseObjTy->getAs<TupleType>()) {
-    StringRef nameStr = memberName.getBaseName().str();
+    StringRef nameStr = memberName.getBaseDeclName().str();
     int fieldIdx = -1;
     // Resolve a number reference into the tuple type.
     unsigned Value = 0;
     if (!nameStr.getAsInteger(10, Value) && Value < tuple->getNumElements()) {
       fieldIdx = Value;
-    } else {
+    } else if (!memberName.isSpecialName()) {
```

Comment:
This keeps the call to `getBaseName` further down from being executed which would crash in case of a special `DeclName`